src/learn/database/index.md

```diff
@@ -0,0 +1,688 @@
+  WHERE id = ?
+  RETURNING id, content, created_at, complete
+`);
+
+function toggleTask(id) {
+  return update_complete.get(id);
+}
+```
+
+This function will flip the `complete` column from `0` to `1`, or from `1` to `0`.
+
+## Integrating a UI
+
+We now have all the functionality for our app contained in the model. All that's left is to build a UI that calls these functions, so that a user can interact with them.
+
+Let's create a Node server that uses our new database. First install `express`:
+
+```shell
+npm install express
+```
+
+Then we'll create the boilerplate we need to get an HTTP server going. First we need a `server.js` file:
+
+```js
+const express = require("express");
+
+const server = express();
+
+server.get("/", (req, res) => {
+  res.send("hello world");
+});
+
+module.exports = server;
+```
+
+and an `index.js` to start the server:
+
+```js
+const server = require("./server.js");
+
+const PORT = process.env.PORT || 3333;
+
+server.listen(PORT, () => console.log(`Listening on http://localhost:${PORT}`));
+```
+
+Check this is working by running `node index.js` in your terminal.
+
+### Submitting new tasks
+
+Now we need a form to add new tasks to our database. Edit `server.js`:
+
+```js
+server.get("/", (req, res) => {
+  const body = /*html*/ `
+    <!doctype html>
+    <form method="POST">
+      <input id="content" name="content" aria-label="New task" required>
+      <button>Add task +</button>
+    </form>
+  `;
+  res.send(body);
+});
+```
+
+Then add a `/ POST` handler to receive the submission a insert the new task:
+
+```js
+const model = require("./model/tasks.js");
+
+server.post("/", express.urlencoded({ extended: false }), (req, res) => {
+  model.createTask(req.body.content);
+  res.redirect("/");
+});
+```
+
+### Rendering tasks
+
+Now that we can insert tasks we need to show them on the page. Edit the `GET /` handler to get the list and render them:
+
+```js
+server.get("/", (req, res) => {
+  const tasks = listTasks();
+  const body = /*html*/ `
+    <!doctype html>
+    <form method="POST">
+      <input id="content" name="content" aria-label="New task" required>
+      <button>Add task +</button>
+      <ul>${tasks.map((t) => `<li>${t.content}</li>`).join("")}</ul>
+    </form>
+  `;
+  res.send(body);
+});
+```
+
+### Updating tasks
+
+We need to be able to toggle and delete each task. This will mean each task `<li>` needs to contain a `<form>` that can send a POST telling the server to either toggle or remove the task.
+
+Let's write a `POST /update` handler _first_. This is what will carry out the changes when a task is toggled or removed. Writing this first will help us know what our form should include.
+
+Our handler is going to need to know two things: which action should it take (toggle or remove), and which task should it update. Both will need to be submitted by the form as part of the request body:
+
+```js
+server.post("/update", express.urlencoded({ extended: false }), (req, res) => {
```

Comment:
removeTask(id)
Should be: model.removeTask    

toggleTask
Should be: model.toggleTask